controllers/httproute_controller.go

```diff
@@ -379,3 +409,45 @@ func isExternalDNSSupported(mgr ctrl.Manager) (bool, error) {
 	}
 	return false, nil
 }
+
+func (r *HTTPRouteReconciler) areBackendRefsIpFamiliesValid(ctx context.Context, httpRoute *gateway_api.HTTPRoute) bool {
```

Comment:
can you return error here not bool, with explicit error message

Reply:
Why is that?

Reply:
Bool value does not tell much. If it's invalid and you return error, this error can be propagated back to the caller, caller will decide what to do. We should propagate error up to the stack and log at the controller level.